File: samples/CameraAccess/CameraAccess/OpenClaw/ToolBridge.swift
import Foundation

@MainActor
protocol ToolBridge: AnyObject {
  var lastToolCallStatus: ToolCallStatus { get set }
  var connectionState: OpenClawConnectionState { get }

  func checkConnection() async
  func resetSession()
  func delegateTask(task: String, toolName: String) async -> ToolResult
}

File: samples/CameraAccessAndroid/app/src/main/java/com/meta/wearable/dat/externalsampleapps/cameraaccess/openclaw/ToolBridge.kt
package com.meta.wearable.dat.externalsampleapps.cameraaccess.openclaw

import kotlinx.coroutines.flow.StateFlow

interface ToolBridge {
    val lastToolCallStatus: StateFlow<ToolCallStatus>
    val connectionState: StateFlow<OpenClawConnectionState>

    suspend fun checkConnection()
    fun resetSession()
    fun setToolCallStatus(status: ToolCallStatus)
    suspend fun delegateTask(task: String, toolName: String = "execute"): ToolResult
}
